### PR TITLE
Map editor: events, rumors

### PR DESCRIPTION
**Pull request by Nordsoft91**

### CODE REVIEW

File: mapeditor/mapsettings/abstractsettings.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * abstractsettings.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#ifndef ABSTRACTSETTINGS_H
+#define ABSTRACTSETTINGS_H
```
Comment (IvanSavenko) [inline mapeditor/mapsettings/abstractsettings.h:11]: We use pragma once, not include guards

File: mapeditor/inspector/inspector.cpp [hunk 2]
```diff
@@ -540,7 +540,7 @@ void Inspector::setProperty(CGArtifact * o, const QString & key, const QVariant
 	{
 		for(auto spell : VLC->spellh->objects)
 		{
-			if(spell->getJsonKey() == value.toString().toStdString())
+			if(spell->getNameTranslated() == value.toString().toStdString())
```
Comment (IvanSavenko) [inline mapeditor/inspector/inspector.cpp:543]: What if there are two spells with same name?

File: mapeditor/mapsettings/abstractsettings.h (hunk 1 above)
Comment (Nordsoft91) [reply mapeditor/mapsettings/abstractsettings.h]: it's automatically generated code by QT

**Comment by dydzio0614** [comment]
will pandora boxes be supported as well?

**Comment by Nordsoft91** [comment]
@dydzio0614 , pandora boxes are supported since 1.2 or even earlier. If it doesn't work, that's one more broken thing in 1.3

**Comment by dydzio0614** [comment]
OK - if you decide to work on hero properties after events then feature-wise editor should be in decent state

**Comment by Nordsoft91** [comment]
that's a plan

File: mapeditor/inspector/inspector.cpp (hunk 2 above)
Comment (Nordsoft91) [reply mapeditor/inspector/inspector.cpp:543]: I'm working on PR with proper way of handling objects via id

**Comment by dydzio0614** [comment]
just to make sure - both map and town timed events are supported in this PR?

**Comment by Nordsoft91** [comment]
no, town events are not supported

**Pull request merged**